packages/common/string.js

```diff
@@ -25,12 +45,94 @@ const unicodeEscape = (str, regex = /[\s\S]/g) => str.replace(regex, unicodeEsca
  * @param {string} oldSubString - The string to replace
  * @param {string} newSubString - The string replacement
  * @returns {string} the modified string
+ *
+ * @static
  */
 function globalReplace(string, oldSubString, newSubString) {
   return string.replace(new RegExp(oldSubString, 'g'), newSubString);
 }
 
+/**
+ * Converts string, as a whole, to lower case just like String#toLowerCase
+ *
+ * @param {string} str - the string to convert
+ * @returns {string} the lower-cased string
+ *
+ * @static
+ */
+const toLower = (str) => str.toLowerCase();
+
+/**
+ * Converts string, as a whole, to upper case just like String#toUpperCase
+ *
+ * @param {string} str - the string to convert
+ * @returns {string} the upper-cased string
+ *
+ * @static
+ */
+const toUpper = (str) => str.toUpperCase();
+
+/**
+ * Tests a regular expression against a String, returning matches
+ *
+ * Produces same output as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match
+ *
+ * This is a curried function - https://lodash.com/docs/4.17.11#curry
+ *
+ * @param {RegExp} regexp - the pattern to match against
+ * @param {string} str - the string to match against
+ * @returns {Array|null}
+ *
+ * @static
+ * @kind function
+ */
+const match = curry((regexp, str) => str.match(regexp), 2);
+
+/**
+ * Tests a regular expression against a string, returning true / false
+ *
+ * This is a curried function - https://lodash.com/docs/4.17.11#curry
+ *
+ * @param {RegExp} regexp - the pattern to match against
+ * @param {string} str - the string to match against
+ * @returns {boolean} true if the pattern matches the string, false otherwise
+ *
+ * @static
+ * @kind function
+ *
+ * @example
+ * const isCapitalized = matches(/^[A-Z]/);
```

Comment:
This example does not match.

Reply:
```
> const stringUtils = require('./string')
undefined
> const isCapitalized = stringUtils.matches(/^[A-Z]/);
undefined
> isCapitalized('Joe');
true
```

Reply:
oh sorry i read this wrong. I found the function a bit confusing.